### PR TITLE
UI: Fix inconsistencies in FindProtocol 

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Fix inconsistencies that might lead to issue or even a crash.

Note: The else branch should not have been taken in custom server scenarios.

### Motivation and Context

Fix a crash reported on discord: https://discord.com/channels/348973006581923840/374636084883095554/1087384798894501939

### How Has This Been Tested?

Switch from a FTL service to custom (with empty server URL) and the encoder warning shown up. Without the fix it doesn't show.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by notr1ch** [comment]
I'm generally not a fan of branches that use the same variable in different conditions like this. Personally I think an early return in the first branch or letting it fall through would be better.

**Review by notr1ch — APPROVED** [review]

File: UI/window-basic-settings-stream.cpp [hunk 1]
```diff
@@ -533,11 +533,15 @@ void OBSBasicSettings::ServiceChanged()
 
 QString OBSBasicSettings::FindProtocol()
 {
-	if (IsCustomService() && !ui->customServer->text().isEmpty()) {
+	if (IsCustomService()) {
+
+		if (ui->customServer->text().isEmpty())
```
Comment (RytoEX) [inline UI/window-basic-settings-stream.cpp:538]: Nit: remove the blank line here.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: UI/window-basic-settings-stream.cpp (hunk 1 above)
Comment (tytan652) [reply UI/window-basic-settings-stream.cpp]: Done

**Review by RytoEX — APPROVED** [review]
Typically, no need to end list items with periods, but not going to block on this.

Looks good to me.

**Pull request merged**